version: '3.8'

services:
  postgres:
    container_name: postgres_container
    image: postgres:16.2
    restart: always
    environment:
      POSTGRES_USER: ${POSTGRES_USER:-pgi}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD:-pgi}
      POSTGRES_DB: ${POSTGRES_DB:-docdatabase}
    expose:
      - "5432"
    ports:
      - "5432:5432"
    volumes:
      - postgres_volume:/var/lib/postgresql/data
      - ./schema.sql:/docker-entrypoint-initdb.dl/schema.sql
    networks:
      - backend
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER:-pgi}"]
      interval: 30s
      timeout: 10s
      retries: 3

  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4:8.5
    restart: always
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_EMAIL:-pgi@planetgroupint.com}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_PASSWORD:-pgi}
      PGADMIN_DEFAULT_ADDRESS: 6000
      PGADMIN_LISTEN_PORT: 6000
    expose:
      - "6000"
    ports:
      - "7000:6000"
    volumes:
      - pgadmin_volume:/var/lib/pgadmin
    networks:
      - backend

networks:
  backend:
    driver: bridge

volumes:
  pgadmin_volume:
  postgres_volume:

